### PR TITLE
Feat/quick grid column footer

### PR DESCRIPTION
**Pull request by ivansicic**
Add column footer to QuickGrid

 - column footer should display column summaries
 - QuickGrid now has columnSummaries prop that takes object having column names as properties and string as values
 - if columnSummaries are not provided, QuickGrid behavior will not be changed

### CODE REVIEW

File: examples/src/QuickGrid.tsx [hunk 1]
```diff
@@ -26,6 +26,13 @@ const gridActions: QuickGridActions = {
     }
 };
 
+const columnSummaries = {
+    Name: 'Craziest: Vinko',
```
Comment (vinkobedek) [inline examples/src/QuickGrid.tsx]: good one :P

File: src/components/QuickGrid/QuickGrid.tsx [hunk 2]
```diff
@@ -142,25 +143,40 @@ export class QuickGridInner extends React.Component<IQuickGridProps, IQuickGridS
...
         if (prevProps.columns !== this.props.columns || prevState.groupBy !== this.state.groupBy || prevState.columnWidths !== this.state.columnWidths) {
             this._grid.recomputeGridSize();
         } else if (this.state.sortDirection !== prevState.sortDirection || this.state.sortColumn !== prevState.sortColumn) {
             this._grid.forceUpdate();
         }
     }
 
-    getGridWidth() {
+
+    getViewportWidth() {
```
Comment (vinkobedek) [inline src/components/QuickGrid/QuickGrid.tsx]: Like the name change to Viewport.

File: src/components/QuickGrid/QuickGridFooter.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    columns: Array<GridColumn>;
+    height: number;
+    rowHeight: number;
+    width: number;
+    rowData: any;
+    scrollLeft: any;
+    onScroll: any;
+}
+
+export interface IGridFooterState {
```
Comment (vinkobedek) [inline src/components/QuickGrid/QuickGridFooter.tsx]: Is this state used? i don't see it used anywhere below.

File: src/components/QuickGrid/QuickGridFooter.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    columnWidths: Array<number>;
+}
+
+export class GridFooter extends React.PureComponent<IGridFooterProps, IGridFooterState> {
+    private _footerGrid: any;
+    private columnMinWidths: Array<number>;
```
Comment (vinkobedek) [inline src/components/QuickGrid/QuickGridFooter.tsx]: Is this used? i don't see it used anywhere below.

File: src/components/QuickGrid/QuickGrid.tsx (hunk 2 above)
Comment (ivansicic) [reply src/components/QuickGrid/QuickGrid.tsx]: Just a step in calculating GridWidth, getGridWidth is using it now.

**Pull request merged**